ReSharper.FSharp/src/FSharp.Psi.Features/src/LanguageService/FSharpElementFactory.fs

```diff
@@ -208,15 +213,52 @@ type FSharpElementFactory(languageService: IFSharpLanguageService, psiModule: IP
 
             expr
             
-        member x.CreateTypedPatInParens(typeSignature: string) : ISynPat =
-            let expr = createLetBinding (sprintf "(%s)" typeSignature)
+        // TODO: Work out whether these factory methods are the simplest they can be
+        member x.CreateParenPat() =
+            let expr = createLetBinding "(())"
             let binding = expr.Bindings |> Seq.exactlyOne
-            binding.HeadPattern  
+            binding.HeadPattern.As<IParenPat>()
             
-        member x.CreateReturnTypeInfo(typeSignature: string) : IReturnTypeInfo =
-            let expr = createLetBinding (sprintf "_ : %s" typeSignature)
-            expr.Bindings.[0].ReturnTypeInfo
-
+        member x.CreateTypedPat (pattern: string, typeUsage: ITypeUsage, spaceBeforeColon: bool) : ITypedPat =
```

Comment:
Could you add an overload taking an existing pattern and putting its copy inside the new typed one? It'd allow reusing patterns you get in IParametersOwnerPat.